rasterio/__init__.py

```diff
@@ -160,11 +161,29 @@ def open(path, mode='r', driver=None, width=None, height=None,
         raise TypeError("invalid driver: {0!r}".format(driver))
     if dtype and not check_dtype(dtype):
         raise TypeError("invalid dtype: {0!r}".format(dtype))
+
+    if 'affine' in kwargs:
+        # DeprecationWarning's are ignored by default
+        with warnings.catch_warnings():
+            warnings.simplefilter('always')
```

Comment:
What about `FutureWarning` which seems to be a bit louder by default than `DeprecationWarning`.